framework/docker/chain_builder.go

```diff
@@ -362,6 +362,16 @@ func (b *ChainBuilder) Build(ctx context.Context) (*Chain, error) {
 		return nil, fmt.Errorf("failed to initialize chain nodes: %w", err)
 	}
 
+	// separate validators and full nodes
+	var validators, fullNodes []*ChainNode
+	for _, node := range nodes {
+		if node.Validator {
+			validators = append(validators, node)
+		} else {
+			fullNodes = append(fullNodes, node)
+		}
+	}
```

Comment:
bug found! It was assigning all the nodes to validators even if we set the type to be full node